src/backend/distributed/commands/type.c

```diff
@@ -669,3 +728,52 @@ LookupNonAssociatedArrayTypeNameOid(ParseState *pstate, const TypeName *typeName
 
 	return typeOid;
 }
+
+
+/*
+ * hasCitusTableDependency checks whether the given type has a dependency
+ * to a Citus table as the type of one of the table's columns
+ */
+static ObjectAddress *
+hasCitusTableDependency(Oid typeId)
+{
```

Comment:
Hmm, we have a very similar function on `src/backend/distributed/test/citus_depended_object.c`, which is more comprehensive because we should search enums a little more carefully (maybe some other types as well). 

Overall, can you check this PR: https://github.com/citusdata/citus/pull/6055/files

And see if we can re-use that code? We have most of the code under src/test because it is only used for testing purposes. But, some parts are in src/backend which might be useful?

Reply:
Apparently `IsCitusDependentObject` checks for objects that the input is dependent on. In this case, we want objects that are dependent on the input. I will rename my function to something like `typeHasDependentCitusTable`.